meps/mep-0003.md

```diff
@@ -0,0 +1,257 @@
+---
+label: MEP0003
+tags:
+  - Active
+title: Inline Options for Roles and Directives
+date: 2025/01/27
+authors:
+  - rowanc1
+  - agoose77
+short_title: Inline Options
+description: An inline syntax for setting options on roles and directives.
+data:
+  discussion: "https://github.com/jupyter-book/myst-enhancement-proposals/pull/28"
+abbreviations:
+  DSL: Domain Specific Language
+---
+
+## Summary
+
+We propose an extension to the MyST Markdown syntax for roles and directives that facilitates inline definition of configuration options.
+
+## Context
+
+There is currently no syntax support for parametrising role definitions[^def] in MyST Markdown. This limitation has led to workarounds such as postfix role name conventions that conceptually group together related roles (e.g. `cite:ps`, `cite:t`), or the use of DSLs in the role body for parsing multiple values (e.g. `` {doc}`title <document.md>` ``). Given that these approaches are both unstandardized and limited in their extensibility, the lack of proper syntax support imposes strong contraints on the utility of roles.
+
+[^def]: Where a role _definition_ is the appearance of the role in a MyST document, rather than the role _declaration_ which specifies its behaviour and interface.
+
+## Proposal
+
+We propose extending the MyST Markdown syntax to support parametrisation of roles and directives. The existing syntax for defining a role and directive of the form
+
+```
+{ROLE-NAME}`ROLE-BODY`
+```
+
+and
+
+````
+```{DIRECTIVE-NAME}
+DIRECTIVE-BODY
+```
+````
+
+will be extended by replacing the `{NAME}` syntax of specifying role/directive names with a more general _named inline attribute_ syntax of the form `{NAME .CLASS #ID KEY=VALUE}`. With this new syntax, it will be possible to directly define directive options in an inline manner, e.g. the following are equivalent:
+
+````
+```{tip}
+:name: my-tip
```

Comment:
Aren't we removing the use of `name` and instead using `label` for all of these? 

https://mystmd.org/guide/cross-references#directive-targets

Reply:
`name` is an alias for label in all cases, I think our docs and MEPs should use label by default, however, both will work.

Reply:
ya I know both of them work, my question is about how we present publicly. Here's the issue where we discussed this for ref:

- https://github.com/jupyter-book/mystmd/issues/1068

Reply:
```suggestion
:label: my-tip
```